api/server.go

```diff
@@ -6,62 +6,53 @@ import (
 	"net/http"
 	"os"
 
-	"github.com/gin-gonic/gin"
-
 	"github.com/UTDNebula/nebula-api/api/configs"
 	"github.com/UTDNebula/nebula-api/api/routes"
+	"github.com/gin-gonic/gin"
+	"github.com/joho/godotenv"
 )
 
 func main() {
-	router := gin.Default()
-
-	// enable cors
-	/*router.Use(cors.New(cors.Config{
-	    AllowMethods:     []string{"GET", "POST", "OPTIONS", "PUT"},
-	    AllowHeaders:     []string{"*"},
-	    ExposeHeaders:    []string{"Content-Length"},
-	    AllowCredentials: false,
-	    AllowAllOrigins:  true,
-	    AllowOriginFunc: func(origin string) bool { return true },
-	    MaxAge:          86400,
-	}))*/
+	loadEnv()
+	loadMongo()
+	serveTraffic()
+}
 
-	router.Use(CORS())
+func loadEnv() {
+	err := godotenv.Load(".env")
+	if err != nil {
+		log.Print("There was no .env file found!")
+	}
+}
 
-	// connect to database
+func loadMongo() {
 	configs.ConnectDB()
+}
+
+func serveTraffic() {
+	router := gin.Default()
+	router.Use(CORS())
 
-	// routes
 	routes.CourseRoute(router)
 	routes.DegreeRoute(router)
 	routes.ExamRoute(router)
 	routes.SectionRoute(router)
 	routes.ProfessorRoute(router)
 
-	//router.OPTIONS("*", CORSOptionsHandler())
-
-	// @DEBUG
-	// router.GET("/", func(c *gin.Context) {
-	//     c.String(http.StatusOK, "Hello World!")
-	// })
-
-	port := os.Getenv("PORT")
-	if port == "" {
+	port, exist := os.LookupEnv("PORT")
```

Comment:
I think we should provide these options through cli options. This might be easier to manage.